src/FlaUI.Core/Input/Wait.cs

```diff
@@ -16,12 +16,16 @@ public static class Wait
         /// Waits for a generic time which was found to be sufficient to allow
         /// input (mouse, keyboard, ...) do be processed
         /// </summary>
-        public static void UntilInputIsProcessed()
+        /// <param name="waitTimeout">An optional timeout. If null is passed, the timeout is infinite.</param>
+        public static void UntilInputIsProcessed(TimeSpan? waitTimeout = null)
         {
             // Let the thread some time to process the system's hardware input queue.
             // For details see this post: http://blogs.msdn.com/b/oldnewthing/archive/2014/02/13/10499047.aspx
             // TODO: Should this be configurable?
-            Thread.Sleep(100);
+            
+            var waitTime = (waitTimeout ?? TimeSpan.FromMilliseconds(-1)).TotalMilliseconds;
```

Comment:
Default should be 100 ms not -1 :)

Reply:
https://msdn.microsoft.com/en-us/library/system.threading.timeout.infinitetimespan(v=vs.110).aspx

Reply:
So in other words, if you don't pass a value it will wait forever. Definitely not what we want as default behavior.